src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.ActionStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final ActionStrategy actionStrategy;
+
+    public ShopServiceImpl(ActionStrategy actionStrategy) {
+        this.actionStrategy = actionStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            ActionHandler handler = actionStrategy.getHandler(transaction.getAction());
```

Comment:
The variable 'handler' should have a more restricted scope. It is only used inside the loop, so consider declaring it inside the loop to enhance encapsulation. This way, you prevent the 'handler' from being accessible outside of its intended scope.